### PR TITLE
fix: improve error handling

### PR DESCRIPTION
**Pull request by setchy**
Improve handling of attributes which may be null or responses that may be null.

I found some of these causing issues when testing locally

### CODE REVIEW

File: src/utils/api-requests.ts [hunk 1]
```diff
@@ -4,7 +4,11 @@ export function apiRequest(
   url: string,
   method: Method,
   data = {},
-): AxiosPromise {
+): AxiosPromise | null {
+  if (!url && !method) {
+    return null;
```
Comment (bmulholland) [inline src/utils/api-requests.ts:9]: Why not create & resolve a promise instead?

File: src/utils/api-requests.test.ts [hunk 2]
```diff
@@ -6,7 +6,18 @@ jest.mock('axios');
 const url = 'https://example.com';
 const method = 'get';
 
-describe('apiRequest', () => {
+describe('utils/api-requests.ts', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('handles null parameters', async () => {
```
Comment (bmulholland) [inline src/utils/api-requests.test.ts:14]: When would this happen? Is it a valid case, or should it be resolved at a higher level?

Comment (setchy) [reply src/utils/api-requests.test.ts:14]: We do handle in most cases checking that the url is not null higher up in the calling business logic.

Perhaps this is overkill.

File: src/utils/helpers.test.ts [hunk 3]
```diff
@@ -445,6 +445,34 @@ describe('utils/helpers.ts', () => {
           `https://github.com/manosim/notifications-test/discussions/612?${mockedNotificationReferrer}#discussioncomment-2300902`,
         );
       });
+
+      it('default to base discussions url when graphql query fails', async () => {
```
Comment (bmulholland) [inline src/utils/helpers.test.ts:449]: Nice one, I like it

File: src/utils/subject.ts [hunk 4]
```diff
@@ -104,16 +104,12 @@ async function getGitifySubjectForDiscussion(
...
   }
 
-  let comments = discussion.comments.nodes;
-  let discussionUser = null;
-
-  if (comments?.length) {
-    discussionUser = getLatestDiscussionComment(comments)?.author.login;
-  }
+  const discussionUser = getLatestDiscussionComment(discussion.comments.nodes)
+    ?.author.login;
```
Comment (bmulholland) [inline src/utils/subject.ts:108]: Does login need a `?` too? Not sure how JS treats this.

```suggestion
    ?.author?.login;
```

File: src/utils/subject.ts [hunk 5]
```diff
@@ -202,10 +198,14 @@ function getWorkflowRunStatus(statusDisplayName: string): CheckSuiteStatus {
 async function getLatestCommentUser(
   notification: Notification,
   token: string,
-): Promise<User> {
+): Promise<User | null> {
+  if (!notification.subject.latest_comment_url) {
+    return null;
```
[the remark this answers is not in the document]
Comment (bmulholland) [reply src/utils/subject.ts:203]: Promise.resolve I think

**Review by bmulholland — APPROVED** [review]
Some questions and suggestions, but overall this is a lot of hardening, thank you!

File: src/utils/subject.ts (hunk 4 above)
Comment (setchy) [reply src/utils/subject.ts:108]: I believe this is ok.  if we get an author object thats not null, then it _should_ contain the login attribute

**Pull request merged**